app/models/components/course/materials_ability_component.rb

```diff
@@ -41,8 +41,7 @@ def allow_students_upload_materials
   end
 
   def allow_staff_manage_materials
-    can :read, Course::Material, material_course_staff_hash
-    can :manage, Course::Material, material_course_staff_hash.deep_merge(concrete_material_hash)
+    can :manage, Course::Material, material_course_staff_hash
```

Comment:
Actually, if you ask me we should only define permissions for concrete `materials` here. Assessments and other models who `has_one_folder` should implement their own permissions. 

This is because currently Assessments can be managed by staff, hence this is convenient - but in the future if we have different permissions (eg. only course_owner can manage a special kind of assessment) - there is some issue where `course_teaching_assistant` is able to delete the assessment materials there.